src/exo/LoopIR_scheduling.py

```diff
@@ -3613,6 +3615,49 @@ def get_typ_mem():
     else:
         new_typ = T.Tensor(shape, False, buf_typ.basetype())
 
+    def check_idx(idx, c, p):
+        assert len(idx) == len(w_exprs)
+        s = get_enclosing_stmt_cursor(c)._node
+        # check win.lo <= e.pt < win.hi
+        for i, w in zip(idx, w_exprs):
+            if not isinstance(w, tuple):
```

Comment:
This is currently buggy. We should also perform a check when `w` isn't a tuple. Consider the following test case: 
```
@proc
def foo():
    x: i8[10, 10, 10]
    for i in seq(0, 10):
        x[i, i, i] = 1.0

foo = stage_mem(foo, foo.find_loop("i"), "x[0:10, 0, 0:10]", "x_tmp")
```